GitUI/UserControls/FileStatusDiffCalculator.cs

```diff
@@ -225,7 +225,7 @@ DiffBranchStatus GetDiffStatus(GitItemStatus f, bool atoBDiff)
             return fileStatusDescs;
 
             static ObjectId GetRevisionOrHead(GitRevision rev, ObjectId headId)
-                => rev.IsArtificial ? headId : rev.ObjectId;
+                => headId is null ? null : rev.IsArtificial ? headId : rev.ObjectId;
```

Comment:
I would rather keep it. I guess the change is not necessary at first glance. It would leave questions due to the mismatch with the function name.

Reply:
If line 113 is changed, this line (228) or 119-1217 to be has to be changed too.
(119-121 may even be clearer but more code).

Reply:
The method name is now out of sync with the implementation. Are calling it from multiple places? If not, may move the added check to the call site?

Reply:
Changed this
Also changed 124/127, not strictly necessary but easier to check completeness.